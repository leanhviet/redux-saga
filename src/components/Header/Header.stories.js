// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'

// Components
import Header from './index'

storiesOf('Header', module).add('Default', () => (
  <MemoryRouter>
    <Header />
  </MemoryRouter>
))
