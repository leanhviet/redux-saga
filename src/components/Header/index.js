import React, { useEffect, useState } from 'react'
import { GoogleLogin, GoogleLogout } from 'react-google-login'
import { withRouter } from 'react-router-dom'

// Themes
import ImageTheme from '../../themes/images'

// Constants
import { USER_KEY } from '../../constants/localStorage'
import ROUTES from '../../constants/routes'

// Utils
import { getStorage, setStorage, removeStorage } from '../../utils/storage'

// Components
import Input from '../Input'
import Profile from '../Profile'

const Header = props => {
  const [isLogin, setLogin] = useState(false)
  const [profile, setmyProfile] = useState({ myProfile: {} })

  // Logout
  const logout = () => {
    setLogin(false)
    removeStorage(USER_KEY)
  }

  // Login success
  const onSuccess = googleUser => {
    setLogin(true)
    setStorage(USER_KEY, googleUser)
    setmyProfile({ myProfile: googleUser.profileObj })
  }

  // Login fail
  const onFailure = error => {
    console.log(error)
  }

  useEffect(() => {
    if (getStorage(USER_KEY)) {
      setLogin(true)
      setmyProfile({ myProfile: getStorage(USER_KEY).profileObj })
    } else {
      setLogin(false)
    }
  }, [])

  const isProfile = isLogin ? (
    <>
      <Profile
        name={profile.myProfile.name}
        src={profile.myProfile.imageUrl}
        subName={profile.myProfile.email}
      />
      <GoogleLogout buttonText="Logout" onLogoutSuccess={logout} />
    </>
  ) : (
    <GoogleLogin
      clientId={process.env.REACT_APP_CLIENT_ID}
      buttonText="Login"
      onSuccess={onSuccess}
      onFailure={onFailure}
      cookiePolicy={'single_host_origin'}
      scope="https://www.googleapis.com/auth/youtube.force-ssl"
    />
  )

  const handleSearchByKeyWord = val =>
    props.history.push(`${ROUTES.SEARCH}/${val}`)

  return (
    <header className="container-fluid header">
      <div className="row header__row">
        <div className="col-md-2 header__row__logo">
          <a href="/">
            <img
              className="logo-link__image img-responsive"
              src={ImageTheme.logo}
              alt="logo"
            />
          </a>
        </div>
        <div className="col-md-6 col-md-push-1 header__row__search">
          <Input
            placeholder="Type to search..."
            handleOnKeyDown={handleSearchByKeyWord}
            className="search"
          />
        </div>
        <div className="col-md-3 col-md-push-1 header__row__profile">
          {isProfile}
        </div>
      </div>
    </header>
  )
}

export default withRouter(Header)
