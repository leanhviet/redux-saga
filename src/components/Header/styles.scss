.header {
  padding-top: 30px;
  height: 130px;
  border: 1px solid $white;

  &__row {
    &__logo img {
      max-width: 60%;
    }

    &__search {
      display: flex;

      & .button {
        margin-left: 30px;
      }
    }

    &__profile {
      display: flex;
      justify-content: flex-end;

      & > .profile {
        margin-right: 20px;
      }
    }
  }
}
