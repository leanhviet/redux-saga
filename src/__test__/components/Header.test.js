// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Header from '../../components/Header'

describe('Header component', () => {
  // Snapshot
  test('should render correctly Header component default props', () => {
    const component = shallow(<Header />)

    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
