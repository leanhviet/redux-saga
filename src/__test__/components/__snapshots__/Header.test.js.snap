// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly Header component default props 1`] = `
<Route>
  <Component />
</Route>
`;
